enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberState.java

```diff
@@ -59,11 +59,9 @@ public HighAvailabilityMemberState masterIsElected( HighAvailabilityMemberContex
                 public HighAvailabilityMemberState masterIsAvailable( HighAvailabilityMemberContext context,
                                                                       InstanceId masterId, URI masterHaURI )
                 {
-//                    assert context.getAvailableMaster() == null;
                     if ( masterId.equals( context.getMyId() ) )
                     {
-                        throw new RuntimeException( "Received a MasterIsAvailable event for my InstanceId while in" +
-                                " PENDING state" );
+                        return PENDING_ELECTION;
```

Comment:
But there is one case where PENDING goes to PENDING, and no election should be triggered. And that it is in PENDING and receives a `masterIsElected` event. Your idea would trigger a new election everytime a new master is elected?